tests/test_api.py

```diff
@@ -230,6 +230,31 @@ def test_restconf_get_if_none_match_namespace():
     assert response.json() == json.loads('{ "testing:enable": false }')
 
 
+def test_restconf_get_if_none_match_put():
+    response = requests.get("{}{}/data/test/settings/enable".format(server_uri, docroot), auth=server_auth, headers=get_restconf_headers)
+    assert response.status_code == 200
+    assert len(response.content) > 0
+    print(json.dumps(response.json(), indent=4, sort_keys=True))
+    etag = response.headers.get("Etag")
+    assert response.json() == json.loads('{ "enable": true }')
+    response = requests.get("{}{}/data/test/settings/enable".format(server_uri, docroot), auth=server_auth, headers={**get_restconf_headers, 'If-None-Match': etag})
```

Comment:
I'm writing new tests using f strings not .format - makes things slightly cleaner to read.